src/interface/variable_pack.hpp

```diff
@@ -142,17 +166,43 @@ VariableFluxPack<T> MakeFluxPack(const vpack_types::VarList<T> &vars,
   ViewOfParArrays<T> f1("MakeFluxPack::f1", fsize);
   ViewOfParArrays<T> f2("MakeFluxPack::f2", fsize);
   ViewOfParArrays<T> f3("MakeFluxPack::f3", fsize);
+  ParArrayND<int> sparse_assoc("MakeFluxPack::sparse_assoc", vsize);
   auto host_view = cv.GetHostMirror();
   auto host_f1 = f1.GetHostMirror();
   auto host_f2 = f2.GetHostMirror();
   auto host_f3 = f3.GetHostMirror();
+  auto host_sp = sparse_assoc.GetHostMirror();
   // add variables to host view
   int vindex = 0;
+  int sparse_start;
+  int sparse_id;
+  std::string sparse_name = "";
   for (const auto &v : vars) {
+    sparse_id = v->metadata().GetSparseId();
+    if (v->IsSet(Metadata::Sparse)) {
+      std::string sparse_trim = v->label();
+      sparse_trim.erase(sparse_trim.find_last_of("_"));
+      if (sparse_name == "") {
+        sparse_name = sparse_trim;
+        sparse_start = vindex;
+      }
+      if (sparse_name != sparse_trim) {
+        vmap->insert(std::pair<std::string, IndexPair>(
+            sparse_name, IndexPair(sparse_start, vindex - 1)));
+        sparse_name = sparse_trim;
+        sparse_start = vindex;
+      }
+
+    } else if (!(sparse_name == "")) {
+      vmap->insert(std::pair<std::string, IndexPair>(
```

Comment:
How do you know that vmap is not still a nullptr here?

Reply:
we don't.  good catch.  fixed.